templates/k8smetricsservicemonitor.go

```diff
@@ -44,7 +44,7 @@ var K8sMetricsServiceMonitorTemplate = promv1.ServiceMonitor{
 				Scheme:        "https",
 				ScrapeTimeout: "1m",
 				Interval:      "2m",
-				HonorLabels:   true,
+				HonorLabels:   false,
```

Comment:
- what'll this do and why this change is required?

Reply:
Since there are two prom instance scraping the same metrics. The ones federated from incluster monitoring.
It creates an error `msg="Error on ingesting out-of-order samples" num_dropped=1` .Which mean two same timeseries are being created with different timestamps . This can be solved by not honoring the labels added by our prometheus instance and instead accepting only the ones federated by incluster-monitoring.More info here
https://www.robustperception.io/debugging-out-of-order-samples/